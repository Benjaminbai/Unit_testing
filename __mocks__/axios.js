
const mockAxios = jest.requireActual('Axios')

export default mockAxios;