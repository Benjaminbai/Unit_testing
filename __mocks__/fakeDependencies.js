jest.mock('react-router-dom', () => {
    const originalModule = jest.requireActual('react-router-dom');

    return {
        __esModule: true,
        ...originalModule,
        useLocation: jest.fn().mockReturnValue({
            pathname: '',
            search: '',
            hash: '',
            state: null,
            key: '',
        }),
        useHistory: () => ({
            push: jest.fn(),
            location: {}
        }),
        Redirect: jest.fn().mockReturnValue({}),
    }
})
window.matchMedia = jest.fn().mockImplementation((query) => {
    const obj = {
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(),
        removeListener: jest.fn(),
    };

    return obj;
})

jest.mock('react', () => ({
    ...jest.requireActual('react'),
    useLayoutEffect: jest.requireActual('react').useEffect
}))