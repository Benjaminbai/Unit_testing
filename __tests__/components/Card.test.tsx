import '../../__mocks__/fakeDependencies'
import React, { Children } from 'react'
import { shallow, mount, render } from 'enzyme'
import { BrowserRouter as Router } from 'react-router-dom'
import Card, { ICard } from '../../components/Card/Card'

const Iprops: ICard = {
    icon: "",
    title: "ben",
    value: "",
    linkTo: "",
    hoverable: false,
}

describe('<Card />', () => {
    test('expect the Card exist', () => {
        const CardComponent = mount(<Router><Card {...Iprops} /></Router>)
        expect(CardComponent.exists()).toBe(true)
    })

    test('expect Card unmount', () => {
        const CardComponent = mount(<Router><Card {...Iprops} /></Router>)
        CardComponent.unmount()
        expect(() => {
            CardComponent.instance()
        }).toThrow()
    })

    // 快照
    test('the source code changed or not', () => {
        const CardComponent = render(<Router><Card {...Iprops} /></Router>)
        expect(CardComponent).toMatchSnapshot()
    })

    test('find title', () => {
        const CardComponent = mount(<Router><Card {...Iprops} /></Router>)
        expect(CardComponent.find('.xui-card').childAt(1).childAt(0).text()).toEqual('ben')
    })
})