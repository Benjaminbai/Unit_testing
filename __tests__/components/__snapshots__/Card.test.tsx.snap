// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> the source code changed or not 1`] = `
<a
  href="/"
>
  <div
    class="xui-card"
  >
    <div
      style="padding:0 10px"
    />
    <div
      style="padding:0 10px"
    >
      <div>
        ben
      </div>
      <div
        style="font-size:24px;font-weight:bold"
      />
    </div>
  </div>
</a>
`;
